x-pack/plugins/ml/public/application/datavisualizer/index_based/index_data_visualizer.tsx

```diff
@@ -5,31 +5,109 @@
  * 2.0.
  */
 
-import React, { FC, Fragment, useEffect, useState } from 'react';
-import { useMlKibana, useTimefilter } from '../../contexts/kibana';
+import React, { FC, Fragment, useEffect, useState, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { useMlKibana, useTimefilter, useMlUrlGenerator } from '../../contexts/kibana';
 import { NavigationMenu } from '../../components/navigation_menu';
 import { HelpMenu } from '../../components/help_menu';
-import type { IndexDataVisualizerViewProps } from '../../../../../data_visualizer/public';
+import { ML_PAGES } from '../../../../common/constants/ml_url_generator';
+import { isFullLicense } from '../../license';
+import { mlNodesAvailable, getMlNodeCount } from '../../ml_nodes_check/check_ml_nodes';
+import { checkPermission } from '../../capabilities/check_capabilities';
+
+import type { ResultLink, IndexDataVisualizerSpec } from '../../../../../data_visualizer/public';
+
+interface GetUrlParams {
+  indexPatternId: string;
+  globalState: any;
+}
+
 export const IndexDataVisualizerPage: FC = () => {
   useTimefilter({ timeRangeSelector: false, autoRefreshSelector: false });
   const {
     services: { docLinks, dataVisualizer },
   } = useMlKibana();
-  const [
-    IndexDataVisualizer,
-    setIndexDataVisualizer,
-  ] = useState<FC<IndexDataVisualizerViewProps> | null>(null);
+  const mlUrlGenerator = useMlUrlGenerator();
+  getMlNodeCount();
+
+  const [IndexDataVisualizer, setIndexDataVisualizer] = useState<IndexDataVisualizerSpec | null>(
+    null
+  );
 
   useEffect(() => {
     if (dataVisualizer !== undefined) {
       const { getIndexDataVisualizerComponent } = dataVisualizer;
       getIndexDataVisualizerComponent().then(setIndexDataVisualizer);
     }
   }, []);
+
+  const links: ResultLink[] = useMemo(
+    () => [
+      {
+        id: 'create_ml_ad_job',
```

Comment:
This link is now shown for non-time based indices, where previously it was hidden (I think). Is it possible to hide it if the index does not contain a timestamp field?

![image](https://user-images.githubusercontent.com/7405507/124462139-4d2d3b00-dd89-11eb-8a8d-0a61545d792b.png)

Reply:
If it isn't easy to detect the presence of a timestamp field, could we change the link to point to the start of the job wizard (select job type) page, as we do for the link on the File data visualizer?

Reply:
Fixed in https://github.com/elastic/kibana/pull/103932/commits/04879a8338c105b30f2662c8a3652234e5263407
Index data visualizer does not show the AD job link for non time series indices.
Also the file data vizualizer does not show the link if the imported index is not time based